modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/runtime/connection/ForwardingSoapClientConnectionProvider.java

```diff
@@ -33,7 +41,11 @@
  *
  * @since 4.0
  */
-public class ForwardingSoapClientConnectionProvider implements PoolingConnectionProvider<ForwardingSoapClient> {
+public class ForwardingSoapClientConnectionProvider implements PoolingConnectionProvider<ForwardingSoapClient>, Lifecycle {
```

Comment:
Still not sure why `PoolingConnectionProvider` is used for WSC, did you look into this?